### PR TITLE
Moved Audio Query Gateway changes to multimodalqna.py

### PR DESCRIPTION
**Pull request by okhleif-10**
## Description

This PR moves our gateway changes from [this PR](https://github.com/opea-project/GenAIComps/pull/974) to the multimodalqna.py file in GenAIExamples

## Issues

Original PR link https://github.com/opea-project/GenAIComps/pull/974

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)
- [x] Others (enhancement, documentation, validation, etc.)

## Dependencies

N/A

## Tests

None

### CODE REVIEW

File: MultimodalQnA/multimodalqna.py [hunk 1]
```diff
@@ -16,7 +17,8 @@
 )
 from comps.cores.proto.docarray import LLMParams
 from fastapi import Request
-from fastapi.responses import StreamingResponse
+from fastapi.responses import JSONResponse, StreamingResponse
+StreamingResponse
```
Comment (dmsuehir) [inline MultimodalQnA/multimodalqna.py]: Copy/paste error?

File: MultimodalQnA/multimodalqna.py [hunk 2]
```diff
@@ -108,73 +126,121 @@ def _handle_message(self, messages):
...
+
+        response = response.json()
+        return response["query"]
+
     async def handle_request(self, request: Request):
         data = await request.json()
         stream_opt = bool(data.get("stream", False))
         if stream_opt:
-            print("[ MultimodalQnAService ] stream=True not used, this has not support streaming yet!")
+            print("[ MultimodalQnAGateway ] stream=True not used, this has not support streaming yet!")
```
Comment (dmsuehir) [inline MultimodalQnA/multimodalqna.py]: IS this an intentional change?

File: MultimodalQnA/multimodalqna.py [hunk 3]
```diff
@@ -108,73 +126,121 @@ def _handle_message(self, messages):
...
+                    initial_inputs = {"prompt": prompt, "image": b64_types["image"][0]}
+                else:
+                    initial_inputs = {"prompt": prompt, "image": ""}
+            else:
+                prompt = messages
+                initial_inputs = {"prompt": prompt, "image": ""}
         else:
-            # print(f"This is the first query, requiring multimodal retrieval. Using multimodal rag megaservice")
-            prompt = prompt_and_image
+            # This is the first query. Ignore image input
```
Comment (dmsuehir) [inline MultimodalQnA/multimodalqna.py:235]: It looks like this is missing a couple of lines that exist in the GenAIComps version of our gateway -
```
                prompt = messages
                initial_inputs = {"prompt": prompt, "image": ""}
```

**Review by dmsuehir — CHANGES_REQUESTED** [review]
Did you rebuild the megaservice container and retest?

File: MultimodalQnA/multimodalqna.py (hunk 1 above)
Comment (okhleif-10) [reply MultimodalQnA/multimodalqna.py]: yes

File: MultimodalQnA/multimodalqna.py (hunk 3 above)
Comment (okhleif-IL) [reply MultimodalQnA/multimodalqna.py:235]: Good catch, not sure how that happened!

File: MultimodalQnA/multimodalqna.py (hunk 2 above)
Comment (okhleif-10) [reply MultimodalQnA/multimodalqna.py]: I did not make this change. So not sure where that came from

Comment (dmsuehir) [reply MultimodalQnA/multimodalqna.py]: This class is named `MultimodalQnAService` was the other one `MultimodalQnAGateway`? Maybe they changed this when they did the move

Comment (okhleif-10) [reply MultimodalQnA/multimodalqna.py]: Oh I see. I can change it to Service then

**Review by dmsuehir — APPROVED** [review]
LGTM as long as it's tested and working.

**Review by mhbuehler — APPROVED** [review]

**Pull request merged**